### ISSUE TITLE
Exception when starting the server :  Address already in use: bind

### ISSUE THREAD

**Issue by Anh-bro** [Melledy/Grasscutter#150]
```
PS F:\UserFolder\Desktop\YuanShen> java -jar grasscutter.jar
[17:41:00] [INFO] Starting `Grasscutter...`
[17:41:00] [INFO] Loaded 300 AvatarTalentDatas.
[17:41:00] [INFO] Loaded 400 AvatarSkillDatas.
[17:41:00] [INFO] Loaded 3794 FetterDatas.
[17:41:00] [INFO] Loaded 73 AvatarSkillDepotDatas.
[17:41:00] [INFO] Loaded 167 MonsterDescribeDatas.
[17:41:00] [INFO] Loaded 1 AvatarCostumeDatas.
[17:41:00] [INFO] Loaded 3032 NpcDatas.
[17:41:00] [INFO] Loaded 66 ReliquaryMainPropDatas.
[17:41:00] [INFO] Loaded 200 MonsterCurveDatas.
[17:41:00] [INFO] Loaded 717 SceneDatas.
[17:41:00] [INFO] Loaded 90 AvatarLevelDatas.
[17:41:00] [INFO] Loaded 6889 ItemDatas.
[17:41:00] [INFO] Loaded 336 AvatarPromoteDatas.
[17:41:00] [INFO] Loaded 335 ReliquaryAffixDatas.
[17:41:00] [INFO] Loaded 90 WeaponLevelDatas.
[17:41:00] [INFO] Loaded 939 WeaponPromoteDatas.
[17:41:00] [INFO] Loaded 9 AvatarFlycloakDatas.
[17:41:00] [INFO] Loaded 6626 GadgetDatas.
[17:41:00] [INFO] Loaded 100 AvatarCurveDatas.
[17:41:00] [INFO] Loaded 42 ReliquarySetDatas.
[17:41:00] [INFO] Loaded 100 WeaponCurveDatas.
[17:41:00] [INFO] Loaded 744 EquipAffixDatas.
[17:41:00] [INFO] Loaded 60 PlayerLevelDatas.
[17:41:00] [INFO] Loaded 101 ReliquaryLevelDatas.
[17:41:00] [INFO] Loaded 2420 ProudSkillDatas.
[17:41:01] [INFO] Loaded 568 MonsterDatas.
[17:41:01] [INFO] Loaded 78 AvatarDatas.
[17:41:01] [ERROR] Relic properties are missing weights! Please check your ReliquaryMainPropExcelConfigData or ReliquaryAffixExcelConfigData files in your ExcelBinOutput folder.
[17:41:01] [INFO] Cluster created with settings {hosts=[localhost:27017], mode=SINGLE, requiredClusterType=UNKNOWN, serverSelectionTimeout='30000 ms'}
[17:41:01] [INFO] Opened connection [connectionId{localValue:2, serverValue:63}] to localhost:27017
[17:41:01] [INFO] Opened connection [connectionId{localValue:1, serverValue:62}] to localhost:27017
[17:41:01] [INFO] Monitor thread successfully connected to server with description ServerDescription{address=localhost:27017, type=STANDALONE, state=CONNECTED, ok=true, minWireVersion=0, maxWireVersion=13, maxDocumentSize=16777216, logicalSessionTimeoutMinutes=30, roundTripTimeNanos=24613000}
[17:41:01] [INFO] Opened connection [connectionId{localValue:3, serverValue:64}] to localhost:27017
[17:41:01] [WARN] [Dispatch] No SSL cert found! Falling back to HTTP server.
Exception in thread "main" java.net.BindException: Address already in use: bind
        at java.base/sun.nio.ch.Net.bind0(Native Method)
        at java.base/sun.nio.ch.Net.bind(Net.java:555)
        at java.base/sun.nio.ch.ServerSocketChannelImpl.netBind(ServerSocketChannelImpl.java:337)
        at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:294)
        at java.base/sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:89)
        at jdk.httpserver/sun.net.httpserver.ServerImpl.<init>(ServerImpl.java:101)
        at jdk.httpserver/sun.net.httpserver.HttpServerImpl.<init>(HttpServerImpl.java:50)
        at jdk.httpserver/sun.net.httpserver.DefaultHttpServerProvider.createHttpServer(DefaultHttpServerProvider.java:35)
        at jdk.httpserver/com.sun.net.httpserver.HttpServer.create(HttpServer.java:150)
        at emu.grasscutter.server.dispatch.DispatchServer.start(DispatchServer.java:182)
        at emu.grasscutter.Grasscutter.main(Grasscutter.java:78)``
```

**Comment by xmplay** [comment]
do you know how to read

**Comment by alt3ri** [comment]
reading issues

**Comment by lunaticwhat** [comment]
tldr moment

**Comment by TheLostTree** [comment]
please its literally in the error itself

**Comment by iTruth** [comment]
just pull the last version and rebuild it

**Comment by aoruLola** [comment]
find **"KeystorePassword": "",** in config.json, the KeystorePassword  is 123456

**Comment by cryingcheems** [comment]
just put [keystore.p12](https://github.com/Grasscutters/Grasscutter/blob/stable/keystore.p12) and grasscutter.jar in the same directory

**Comment by iTruth** [comment]
> just put [keystore.p12](https://github.com/Grasscutters/Grasscutter/blob/stable/keystore.p12) and grasscutter.jar in the same directory

this problem is mostly due to wrong password, so just delete the config.json and rebuild

**Comment by cryingcheems** [comment]
> > just put [keystore.p12](https://github.com/Grasscutters/Grasscutter/blob/stable/keystore.p12) and grasscutter.jar in the same directory
> 
> this problem is mostly due to wrong password, so just delete the config.json and rebuild

What you said is also a possibility, but I just encountered the same error because I didn't put the files in the same directory

**Comment by mrkaria** [comment]
1.netstat -ano|findstr 443，Find the process occupying port 443 and kill it,maybe a vmware problem
![image](https://user-images.githubusercontent.com/41931603/165003167-7fdad1fb-ef11-4aa2-83fd-60c0ce33f34d.png)
modify it to another port number
2.**download new p12 file** [](https://github.com/Grasscutters/Grasscutter/blob/development/keystore.p12) ,set the **KeystorePassword** value in **config.json** to 123456
![image](https://user-images.githubusercontent.com/41931603/165003122-2cec69d2-249f-424f-98e5-5c1c7aa8ed7c.png)

**Comment by 1462903025** [comment]
me too 

> [root@VM-16-8-centos ys]# java -jar grasscutter.jar 
[03:31:05] [INFO] Starting Grasscutter...
[03:31:06] [INFO] Loaded 300 AvatarTalentDatas.
[03:31:06] [INFO] Loaded 400 AvatarSkillDatas.
[03:31:06] [INFO] Loaded 3794 FetterDatas.
[03:31:06] [INFO] Loaded 73 AvatarSkillDepotDatas.
[03:31:06] [INFO] Loaded 167 MonsterDescribeDatas.
[03:31:06] [INFO] Loaded 1 AvatarCostumeDatas.
[03:31:06] [INFO] Loaded 3032 NpcDatas.
[03:31:06] [INFO] Loaded 66 ReliquaryMainPropDatas.
[03:31:06] [INFO] Loaded 200 MonsterCurveDatas.
[03:31:06] [INFO] Loaded 717 SceneDatas.
[03:31:06] [INFO] Loaded 90 AvatarLevelDatas.
[03:31:06] [INFO] Loaded 6889 ItemDatas.
[03:31:06] [INFO] Loaded 336 AvatarPromoteDatas.
[03:31:06] [INFO] Loaded 335 ReliquaryAffixDatas.
[03:31:06] [INFO] Loaded 90 WeaponLevelDatas.
[03:31:06] [INFO] Loaded 939 WeaponPromoteDatas.
[03:31:06] [INFO] Loaded 9 AvatarFlycloakDatas.
[03:31:07] [INFO] Loaded 6626 GadgetDatas.
[03:31:07] [INFO] Loaded 100 AvatarCurveDatas.
[03:31:07] [INFO] Loaded 42 ReliquarySetDatas.
[03:31:07] [INFO] Loaded 100 WeaponCurveDatas.
[03:31:07] [INFO] Loaded 744 EquipAffixDatas.
[03:31:07] [INFO] Loaded 60 PlayerLevelDatas.
[03:31:07] [INFO] Loaded 101 ReliquaryLevelDatas.
[03:31:07] [INFO] Loaded 2420 ProudSkillDatas.
[03:31:07] [INFO] Loaded 568 MonsterDatas.
[03:31:07] [INFO] Loaded 78 AvatarDatas.
[03:31:08] [ERROR] Relic properties are missing weights! Please check your ReliquaryMainPropExcelConfigData or ReliquaryAffixExcelConfigData files in your ExcelBinOutput folder.
[03:31:08] [INFO] Cluster created with settings {hosts=[localhost:27017], mode=SINGLE, requiredClusterType=UNKNOWN, serverSelectionTimeout='30000 ms'}
[03:31:08] [INFO] Opened connection [connectionId{localValue:1, serverValue:20}] to localhost:27017
[03:31:08] [INFO] Monitor thread successfully connected to server with description ServerDescription{address=localhost:27017, type=STANDALONE, state=CONNECTED, ok=true, minWireVersion=0, maxWireVersion=9, maxDocumentSize=16777216, logicalSessionTimeoutMinutes=30, roundTripTimeNanos=52234684}
[03:31:08] [INFO] Opened connection [connectionId{localValue:2, serverValue:21}] to localhost:27017
[03:31:08] [INFO] Opened connection [connectionId{localValue:3, serverValue:22}] to localhost:27017
Exception in thread "main" java.net.BindException: Address already in use
        at java.base/sun.nio.ch.Net.bind0(Native Method)
        at java.base/sun.nio.ch.Net.bind(Net.java:555)
        at java.base/sun.nio.ch.ServerSocketChannelImpl.netBind(ServerSocketChannelImpl.java:337)
        at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:294)
        at java.base/sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:89)
        at jdk.httpserver/sun.net.httpserver.ServerImpl.<init>(ServerImpl.java:104)
        at jdk.httpserver/sun.net.httpserver.HttpsServerImpl.<init>(HttpsServerImpl.java:50)
        at jdk.httpserver/sun.net.httpserver.DefaultHttpServerProvider.createHttpsServer(DefaultHttpServerProvider.java:39)
        at jdk.httpserver/com.sun.net.httpserver.HttpsServer.create(HttpsServer.java:95)
        at emu.grasscutter.server.dispatch.DispatchServer.start(DispatchServer.java:166)
        at emu.grasscutter.Grasscutter.main(Grasscutter.java:78)

**Comment by memetrollsXD** [comment]
In @1462903025's case, it's because something is already running in one of these following ports: 80, 443, 8888 or 22102

**Comment by 1462903025** [comment]
> 

Is there any way to modify the port because I also run other web applications @memetrollsXD

**Comment by 1462903025** [comment]
No ports is occupied this time 

> [root@VM-16-8-centos ys]# netstat -lnp|grep 80
udp6       0      0 fe80::5054:ff:fe5f::123 :::*                                655/ntpd            
[root@VM-16-8-centos ys]# java -jar grasscutter.jar 
[04:15:12] [INFO] Starting Grasscutter...
[04:15:13] [INFO] Loaded 300 AvatarTalentDatas.
[04:15:13] [INFO] Loaded 400 AvatarSkillDatas.
[04:15:13] [INFO] Loaded 3794 FetterDatas.
[04:15:13] [INFO] Loaded 73 AvatarSkillDepotDatas.
[04:15:13] [INFO] Loaded 167 MonsterDescribeDatas.
[04:15:13] [INFO] Loaded 1 AvatarCostumeDatas.
[04:15:13] [INFO] Loaded 3032 NpcDatas.
[04:15:13] [INFO] Loaded 66 ReliquaryMainPropDatas.
[04:15:13] [INFO] Loaded 200 MonsterCurveDatas.
[04:15:13] [INFO] Loaded 717 SceneDatas.
[04:15:13] [INFO] Loaded 90 AvatarLevelDatas.
[04:15:13] [INFO] Loaded 6889 ItemDatas.
[04:15:13] [INFO] Loaded 336 AvatarPromoteDatas.
[04:15:13] [INFO] Loaded 335 ReliquaryAffixDatas.
[04:15:13] [INFO] Loaded 90 WeaponLevelDatas.
[04:15:13] [INFO] Loaded 939 WeaponPromoteDatas.
[04:15:13] [INFO] Loaded 9 AvatarFlycloakDatas.
[04:15:13] [INFO] Loaded 6626 GadgetDatas.
[04:15:13] [INFO] Loaded 100 AvatarCurveDatas.
[04:15:13] [INFO] Loaded 42 ReliquarySetDatas.
[04:15:13] [INFO] Loaded 100 WeaponCurveDatas.
[04:15:13] [INFO] Loaded 744 EquipAffixDatas.
[04:15:13] [INFO] Loaded 60 PlayerLevelDatas.
[04:15:13] [INFO] Loaded 101 ReliquaryLevelDatas.
[04:15:14] [INFO] Loaded 2420 ProudSkillDatas.
[04:15:14] [INFO] Loaded 568 MonsterDatas.
[04:15:14] [INFO] Loaded 78 AvatarDatas.
[04:15:14] [ERROR] Relic properties are missing weights! Please check your ReliquaryMainPropExcelConfigData or ReliquaryAffixExcelConfigData files in your ExcelBinOutput folder.
[04:15:14] [INFO] Cluster created with settings {hosts=[localhost:27017], mode=SINGLE, requiredClusterType=UNKNOWN, serverSelectionTimeout='30000 ms'}
[04:15:14] [INFO] Opened connection [connectionId{localValue:1, serverValue:23}] to localhost:27017
[04:15:14] [INFO] Monitor thread successfully connected to server with description ServerDescription{address=localhost:27017, type=STANDALONE, state=CONNECTED, ok=true, minWireVersion=0, maxWireVersion=9, maxDocumentSize=16777216, logicalSessionTimeoutMinutes=30, roundTripTimeNanos=40546591}
[04:15:14] [INFO] Opened connection [connectionId{localValue:2, serverValue:22}] to localhost:27017
[04:15:14] [INFO] Opened connection [connectionId{localValue:3, serverValue:24}] to localhost:27017
[04:15:15] [INFO] [Dispatch] Dispatch server started on port 443
[04:15:15] [INFO] Registered 86 PacketHandlers
[04:15:15] [INFO] Banners successfully loaded.
Exception in thread "main" org.reflections.ReflectionsException: Scanner TypeAnnotationsScanner was not configured
        at org.reflections.Store.get(Store.java:39)
        at org.reflections.Store.get(Store.java:61)
        at org.reflections.Store.get(Store.java:46)
        at org.reflections.Reflections.getTypesAnnotatedWith(Reflections.java:429)
        at org.reflections.Reflections.getTypesAnnotatedWith(Reflections.java:416)
        at emu.grasscutter.command.CommandMap.scan(CommandMap.java:147)
        at emu.grasscutter.command.CommandMap.<init>(CommandMap.java:19)
        at emu.grasscutter.server.game.GameServer.<init>(GameServer.java:60)
        at emu.grasscutter.Grasscutter.main(Grasscutter.java:80)

**Comment by memetrollsXD** [comment]
See [#148](https://github.com/Grasscutters/Grasscutter/issues/148)

**Comment by 1462903025** [comment]
> 见 [#148](https://github.com/Grasscutters/Grasscutter/issues/148)

I found a problem. I use jdk-18, which is normal after replacement

**Comment by 1462903025** [comment]
> 见 [#148](https://github.com/Grasscutters/Grasscutter/issues/148)

THX BRO After changing the JDK version, it is normal

**Comment by FumoLux** [comment]
```
[14:25:15] [INFO] Welcome to Grasscutter
[14:25:17] [INFO] To proper exit this console, use [Ctrl + C] and enter N not Y.
[14:25:17] [INFO]
[14:25:17] [INFO] Initializing...
[14:25:17] [INFO] Starting proxy daemon...
[14:25:17] [INFO] Waiting for CA certificate generation...
[14:25:18] [INFO] Adding CA certificate to store...
[14:25:18] [INFO] Setting up network proxy...
[14:25:18] [INFO] Starting MongoDB daemon...
[14:25:18] [INFO] Starting server...
[14:25:20] [[34mINFO[0;39m] Starting Grasscutter...
[14:25:20] [[34mINFO[0;39m] Loaded 300 AvatarTalentDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 400 AvatarSkillDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 3794 FetterDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 73 AvatarSkillDepotDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 167 MonsterDescribeDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 1 AvatarCostumeDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 3032 NpcDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 66 ReliquaryMainPropDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 200 MonsterCurveDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 717 SceneDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 90 AvatarLevelDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 6889 ItemDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 336 AvatarPromoteDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 335 ReliquaryAffixDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 90 WeaponLevelDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 939 WeaponPromoteDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 9 AvatarFlycloakDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 6626 GadgetDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 100 AvatarCurveDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 42 ReliquarySetDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 100 WeaponCurveDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 744 EquipAffixDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 60 PlayerLevelDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 101 ReliquaryLevelDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 2420 ProudSkillDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 568 MonsterDatas.
[14:25:20] [[34mINFO[0;39m] Loaded 78 AvatarDatas.
[14:25:20] [[1;31mERROR[0;39m] Relic properties are missing weights! Please check your ReliquaryMainPropExcelConfigData or ReliquaryAffixExcelConfigData files in your ExcelBinOutput folder.
[14:25:21] [[34mINFO[0;39m] Cluster created with settings {hosts=[localhost:27017], mode=SINGLE, requiredClusterType=UNKNOWN, serverSelectionTimeout='30000 ms'}
[14:25:21] [[34mINFO[0;39m] Opened connection [connectionId{localValue:2, serverValue:2}] to localhost:27017
[14:25:21] [[34mINFO[0;39m] Opened connection [connectionId{localValue:1, serverValue:1}] to localhost:27017
[14:25:21] [[34mINFO[0;39m] Monitor thread successfully connected to server with description ServerDescription{address=localhost:27017, type=STANDALONE, state=CONNECTED, ok=true, minWireVersion=0, maxWireVersion=13, maxDocumentSize=16777216, logicalSessionTimeoutMinutes=30, roundTripTimeNanos=10867800}
[14:25:21] [[34mINFO[0;39m] Opened connection [connectionId{localValue:3, serverValue:3}] to localhost:27017
[14:25:21] [[31mWARN[0;39m] [Dispatch] No SSL cert found! Falling back to HTTP server.
Exception in thread "main" java.net.BindException: Address already in use: bind
        at sun.nio.ch.Net.bind0(Native Method)
        at sun.nio.ch.Net.bind(Unknown Source)
        at sun.nio.ch.Net.bind(Unknown Source)
        at sun.nio.ch.ServerSocketChannelImpl.bind(Unknown Source)
        at sun.nio.ch.ServerSocketAdaptor.bind(Unknown Source)
        at sun.net.httpserver.ServerImpl.<init>(Unknown Source)
        at sun.net.httpserver.HttpServerImpl.<init>(Unknown Source)
        at sun.net.httpserver.DefaultHttpServerProvider.createHttpServer(Unknown Source)
        at com.sun.net.httpserver.HttpServer.create(Unknown Source)
        at emu.grasscutter.server.dispatch.DispatchServer.start(DispatchServer.java:182)
        at emu.grasscutter.Grasscutter.main(Grasscutter.java:78)
[14:25:22] [INFO] Server stopped
[14:25:38] [INFO] Shutting down MongoDB daemon...
[14:25:39] [INFO] Restoring network settings...
[14:25:39] [INFO] Shutting down proxy daemon...
[14:25:40] [INFO] Removing CA certificate...
```
I have the same problem and I've tried changing the port, but there is nothing change.
Both of the keystore.p12 and grasscutter.jar are in the same directory

**Comment by Shimogawa** [comment]
> find **"KeystorePassword": "",** in config.json, the KeystorePassword is 123456

Solved by this one. Thanks!
